type-annotator-scanner/src/test/java/edu/ucr/cs/riple/scanner/FieldTrackerTest.java

```diff
@@ -65,4 +65,73 @@ public void BasicTest() {
         .setExpectedOutputs(new TrackerNodeDisplay("edu.ucr.A", "bar()", "edu.ucr.Other", "foo"))
         .doTest();
   }
+
+  @Test
+  public void fieldDeclaredInInnerClassInMethod() {
+    tester
+        .addSourceLines(
+            "edu/ucr/A.java",
+            "package edu.ucr;",
+            "public class A {",
+            "   Other other = new Other();",
+            "   public void bar(){",
+            "      class Foo {;",
+            "          Object baz = other.foo;",
+            "      }",
+            "   }",
+            "}",
+            "class Other {",
+            "   Object foo;",
+            "}")
+        .setExpectedOutputs(
+            new TrackerNodeDisplay("edu.ucr.A$1Foo", "baz", "edu.ucr.Other", "foo"),
+            new TrackerNodeDisplay("edu.ucr.A$1Foo", "baz", "edu.ucr.Other", "foo"),
+            new TrackerNodeDisplay("edu.ucr.A$1Foo", "baz", "edu.ucr.A", "other"))
+        .doTest();
+  }
+
+  @Test
+  public void fieldDeclaredRegionComputationAllCases() {
+    tester
+        .addSourceLines(
+            "edu/ucr/A.java",
+            "package edu.ucr;",
+            "public class A {",
+            "   B b = new B();",
+            "   Object f0 = b.foo;",
+            "   Object f1 = B.staticFoo;",
+            "   Object f2 = B.bar();",
+            "   Object f3 = b.c.val;",
+            "   Object f4 = B.staticC.val;",
+            "}",
+            "class B {",
+            "   C c = new C();",
+            "   static C staticC = new C();",
+            "   Object foo;",
+            "   static Object staticFoo;",
+            "   static Object bar() {",
+            "       return new Object();",
+            "   }",
+            "}",
+            "class C {",
+            "   Object val;",
+            "   static Object get() {",
+            "       return new Object();",
+            "   }",
+            "}")
+        .setExpectedOutputs(
+            new TrackerNodeDisplay("edu.ucr.A", "f0", "edu.ucr.B", "foo"),
+            new TrackerNodeDisplay("edu.ucr.A", "f0", "edu.ucr.B", "foo"),
```

Comment:
Methods `matchIdentifier` and `matchMemberSelect` might produce the same output, I just thought it might not worth it to keep all the outputs in a set and check for every output if it is already serialized as they will be deserialized and converted into a set in `core`. Also I didn't want to change the test infra in this PR. Please let me know if you think I should change the infra or make a list and prevent duplicates serialize in output.

Reply:
Makes sense to me, just wanted to understand this. I do think eventually you might want tests to not need this repetition, but it's low priority and not needed for this PR.